packages/httpsnippet-client-api/src/index.ts

```diff
@@ -169,7 +170,20 @@ const client: Client<APIOptions> = {
     Array.from(Object.entries(operationSlugs)).forEach(([param, value]) => {
       // The keys in `operationSlugs` will always be prefixed with a `:` in the `oas` library so
       // we can safely do this substring here without asserting this context.
-      metadata[param.substring(1)] = value;
+      const cleanedParam = param.substring(1);
+
+      // If our incoming path slug out of `oas.findOperation()` has been sanitized and is missing
+      // a hyphen, but there is a parameter in the OpenAPI definition that matches what our
+      // hyphen-less slug is then we should use that for the snippet.
+      const unsanitizedParam = operationPathParameters.find(p => {
```

Comment:
This thing is/was only a problem on path parameters because we grab these params out of `oas.findOperation()` which uses the `path-to-regexp` library and munges our path params a bit.